java/vector/src/main/codegen/templates/FixedValueVectors.java

```diff
@@ -324,44 +335,51 @@ private void incrementAllocationMonitor() {
   public final class Accessor extends BaseDataValueVector.BaseAccessor {
     @Override
     public int getValueCount() {
-      return data.writerIndex() / ${type.width};
+      return data.writerIndex() / TYPE_WIDTH;
     }
 
     @Override
     public boolean isNull(int index){
       return false;
     }
 
-    <#if (type.width > 8 || minor.class == "IntervalDay")>
+    <#if (type.width < 0 || type.width > 8 || minor.class == "IntervalDay")>
     public ${minor.javaType!type.javaType} get(int index) {
-      return data.slice(index * ${type.width}, ${type.width});
+      <#if (minor.class == "FixedSizeBinary")>
```

Comment:
I think it's cleaner if we get `FixedSizeBinary` into it's own top level "if":

```
<#if (type.width) < 0 && minor.class == "FixedSizeBinary">
public ${minor.javaType!type.javaType} get(int index) {
   ...
}
```
Since it doesn't share logic with the original `get` method